File: misc/ipa.py
#!/usr/bin/python3

# Yes, I could use 'ifconfig' or 'ip addr' but 99% of the time ONLY want the interface name and IP address.
# Output looks like:
# interface: ipaddress

import netifaces

string = str()
for interface in netifaces.interfaces():
    addrs = netifaces.ifaddresses(interface)
    try:
        for item in addrs[netifaces.AF_INET]:
            string += f"{interface}: {item.get('addr')}\n"
    except Exception as e:
        pass

print(string)

File: misc/ipa.go
package main

import (
	"fmt"
	"net"
)

func main() {
	iface, _ := net.Interfaces()

	for _, i := range iface {
		addrs, _ := i.Addrs()
		if addrs != nil {
			fmt.Print(i.Name)
			fmt.Print(": ")
			fmt.Println(addrs[0])
		}
	}
}
